src/main/java/com/flansmod/client/model/RenderGun.java

```diff
@@ -263,6 +264,32 @@ private void renderWeaponSwitchMovement(GunAnimations animations) {
 		GL11.glRotatef(startAngles.z + (endAngles.z - startAngles.z) * interp, 0f, 0f, 1f);
 	}
 
+	private void renderWeaponSprintMovement(GunAnimations animations, ModelGun model, GunType gunType) {
+		Vector3f defaultTranslate = new Vector3f(0, 0F, -0.2);
+		Vector3f defaultRotation = new Vector3f(-15F, 45F, -10F);
+
+		Vector3f configuredTranslate = new Vector3f(model.sprintStanceTranslate.x, model.sprintStanceTranslate.y, model.sprintStanceTranslate.z);
+		Vector3f configuredRotation = new Vector3f(model.sprintStanceRotate.x, model.sprintStanceRotate.y, model.sprintStanceRotate.z);
+
+		float progress = (animations.runningStanceAnimationProgress + smoothing) / animations.runningStanceAnimationLength;
+		if (animations.runningStanceAnimationProgress == animations.runningStanceAnimationLength)
+			progress = 1;
+
+		if (FlansMod.enableRandomSprintStance) {
+			animations.updateSprintStance(gunType.getShortName());
+			defaultRotation = animations.sprintingStance;
+		}
+
+		GL11.glTranslatef(defaultTranslate.x * progress, defaultTranslate.y * progress, defaultTranslate.z * progress);
+		GL11.glTranslatef(configuredTranslate.x * progress, configuredTranslate.y * progress, configuredTranslate.z * progress);
+		GL11.glRotatef(defaultRotation.x * progress, 1f, 0f, 0f);
+		GL11.glRotatef(defaultRotation.y * progress, 0f, 1f, 0f);
+		GL11.glRotatef(defaultRotation.z * progress, 0f, 0f, 1f);
+		GL11.glRotatef(configuredRotation.x * progress, 1f, 0f, 0f);
```

Comment:
hm, so `model.sprintStanceRotate`these act as offsets to the default I think?

Reply:
Yeah. I figured having the default as a starting point would be a lot easier but open to making it not apply them both if preferred

Reply:
My only concern is that the rotation is now with respect to the gun in the default position rather than relative to the player, so people might find it hard to modify

Reply:
I suppose you could add the rotations together, then it's still wrt the player?

Reply:
I think Ill just completely separate it to avoid any weirdness

Reply:
This approach works fine, but I want to make sure we're on the same page with what it means for the rotations:

Before any rotation, the X axis points forwards in the direction of the gun, Y is almost vertical, and Z is to the side.

Then the default stance rotation happens, in each of these axis. Now the X axis is pointing in the direction of the gun - no longer forward. Y axis is altered, as is Z.

The offset rotation that someone has configured will now rotate these axis - not the original ones. That could be confusing imo.

The alternative is glRotateF(default.x + offset.x) for each component so all of the offset rotations act relative to where players expect the gun to be.

I'm pretty sure that's what would happen, but there's a chance I'm wrong. It's not all that important here, but it's either something that needs to be documented somewhere and kept in mind for other configuration 

Note: not the case with glTranslateF, because doing that twice is the same as adding the vectors together.

Reply:
I changed it to be either or, this way a configured stance isnt dependent on anything else. Coulda been problematic if a gun was configured against default and then seeded positions were turned on.

Reply:
hm, I get what you mean ye. That's alright. Just bear it in mind :)